### PR TITLE
Fix compiler warnings in lv_draw_img.c

### PR DESCRIPTION
**Pull request by BrianPugh**

### CODE REVIEW

File: src/lv_draw/lv_draw_img.c [hunk 1]
```diff
@@ -447,7 +448,7 @@ lv_img_dsc_t *lv_img_buf_alloc(lv_coord_t w, lv_coord_t h, lv_img_cf_t cf)
         lv_mem_free(dsc);
         return NULL;
     }
-    memset(dsc->data, 0, dsc->data_size);
+    memset((uint8_t *)dsc->data, 0, dsc->data_size);
```
Comment (embeddedt) [inline src/lv_draw/lv_draw_img.c:451]: Is this cast necessary? [`memset`](https://pubs.opengroup.org/onlinepubs/7908799/xsh/memset.html) should take `void *` as the first parameter.

Comment (BrianPugh) [reply src/lv_draw/lv_draw_img.c:451]: `dsc->data` is declared as a `const`, so either that should change or we have to cast it here as not a `const`.

Comment (embeddedt) [reply src/lv_draw/lv_draw_img.c:451]: > `dsc->data` is declared as a `const`

Makes sense. This is fine.

**Comment by embeddedt** [comment]
Thanks.

**Comment by kisvegabor** [comment]
I wonder why we haven't got warnings for it.

Anyway, thank you for the fix!

**Comment by embeddedt** [comment]
@kisvegabor I assume it's because the warning about casting `const` to non-`const` or vice versa needs to be enabled manually.

**Pull request merged**